tests/pds/pds_storagesize_increase_test.go

```diff
@@ -0,0 +1,901 @@
+							Secure:         false,
+							VolGroups:      false,
+						})
+						stIDs = append(stIDs, stConfigModel.GetId())
+						resIds = append(resIds, resConfigModel.GetId())
+						depList = append(depList, deployment)
+						log.InfoD("Initial deployment ID- %v", deployment.GetId())
+						dataserviceID, _ := dsTest.GetDataServiceID(ds.Name)
+						stepLog = "Update the resource/storage template with increased storage size"
+						Step(stepLog, func() {
+							newTemplateName := "autoTemp-" + strconv.Itoa(rand.Int())
+							updatedTemplateConfig := controlplane.Templates{
+								CpuLimit:       *resConfigModel.CpuLimit,
+								CpuRequest:     *resConfigModel.CpuRequest,
+								DataServiceID:  dataserviceID,
+								MemoryLimit:    *resConfigModel.MemoryLimit,
+								MemoryRequest:  *resConfigModel.MemoryRequest,
+								Name:           newTemplateName,
+								StorageRequest: params.StorageConfigurations.NewStorageSize,
+								FsType:         *stConfigModel.Fs,
+								ReplFactor:     *stConfigModel.Repl,
+								Provisioner:    *stConfigModel.Provisioner,
+								Secure:         false,
+								VolGroups:      false,
+							}
+							stConfigModelUpdated, resConfigModelUpdated, err = controlPlane.CreateCustomResourceTemplate(tenantID, updatedTemplateConfig)
+							log.FailOnError(err, "Unable to update template")
+							log.InfoD("Successfully updated the template with ID- %v", resConfigModelUpdated.GetId())
+							newResourceTemplateID = resConfigModelUpdated.GetId()
+							newStorageTemplateID = stConfigModelUpdated.GetId()
+							stIDs = append(stIDs, newStorageTemplateID)
+							resIds = append(resIds, newResourceTemplateID)
+						})
+						stepLog = "Apply updated template to the dataservice deployment"
+						Step(stepLog, func() {
+							log.InfoD(stepLog)
+							if appConfigID == "" {
+								appConfigID, err = controlPlane.GetAppConfTemplate(tenantID, ds.Name)
+								log.FailOnError(err, "Error while fetching AppConfigID")
+							}
+							updatedDeployment, err = dsTest.UpdateDataServices(deployment.GetId(),
+								appConfigID, deployment.GetImageId(),
+								int32(ds.Replicas), newResourceTemplateID, params.InfraToTest.Namespace)
+							log.FailOnError(err, "Error while updating dataservices")
+							Step("Validate Deployments after template update", func() {
+								err = dsTest.ValidateDataServiceDeployment(updatedDeployment, namespace)
+								log.FailOnError(err, "Error while validating dataservices")
+								log.InfoD("Data-service: %v is up and healthy", ds.Name)
+								updatedDepList = append(updatedDepList, updatedDeployment)
+								updatedPvcSize, err = GetVolumeCapacityInGB(namespace, updatedDeployment)
+								log.InfoD("Updated Storage Size is- %v", updatedPvcSize)
+								dsEntity = restoreBkp.DSEntity{
+									Deployment: updatedDeployment,
+								}
+							})
+							stepLog = "Validate Workload is running after storage resize"
+							Step(stepLog, func() {
```

Comment:
Validate deployment will just check the deployment is up or not. It will not actually validate the workload in running or not. 
You need to run workload again after SSI